flink-streaming-java/src/test/java/org/apache/flink/streaming/util/TestHarnessUtil.java

```diff
@@ -139,6 +139,7 @@ public static void assertNoLateRecords(Iterable<Object> elements) {
 
         testHarness.processElements(
                 input.stream().map(StreamRecord::new).collect(Collectors.toList()));
+        testHarness.prepareSnapshotPreBarrier(1);
```

Comment:
Nit: Why was this changed needed?

Reply:
Since writer+committer are merged, we now need to simulate more of the actual calls. The writer creates the committables on `prepareSnapshotPreBarrier` and pass it to the committer which then does it's actual work on `snapshotState`.